lib/housing-wanted/housing-wanted.controllers.js

```diff
@@ -1,7 +1,118 @@
+    const Log = require('../log/log.controllers')(models).lib;
   
-    return {}
+    return {
+      postHousingWanted: async function(request, h){
+        /*
+          # Require Account for submitting fields other than address
+        */
+        let payload = request.payload;
+        let account;
+        try {
+         account = await accountLib.getAccount(request.auth.credentials);
+        } catch (e) {
+         throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+       }
+
+        if (account === null) {
+          throw Boom.badRequest('Must have an Account to create a housing wanted post');
+        }
+
+        /* Restrict AuthorId to same user, or SuperAdmin */
+        if (payload.AuthorId) {
+          // If AuthorId is set then we have already passed auth and account lookup
+          // This is because if AuthorId is set then not only quickInfo is set
+          if (
+            payload.AuthorId !== account.id  &&
+            request.auth.credentials.subjectId !== process.env.SUPER_ADMIN
+          ) {
+            throw Boom.badRequest('HousingWanted.AuthorId must match your Account ID or you must be a Super Admin');
+          }
+        }
+
+        // bathrooms validation
+        let hasBathrooms = false;
+        let bathrooms = Number(request.payload.bathrooms);
+        if (bathrooms) {
+          hasBathrooms = true;
+        }
+
+        // number of bathrooms should be set
+        // return true if bathrooms object has specified property
+        if (request.payload.hasOwnProperty("bathrooms")){
+          hasBathrooms = true;
+        }
+
+        // bathrooms can be in .25 increments
+        if (hasBathrooms && (!bathroomsValidIncrement(bathrooms)) ) {
+          throw Boom.badData('bathrooms must be in .25 increments');
+        }
+
+        // bedrooms validation
+        let hasBedrooms = false;
+        let bedrooms = Number(request.payload.bedrooms);
+        // return true if bedrooms object has specified property
+        if (request.payload.hasOwnProperty("bedrooms")){
+          hasBedrooms = true;
+        }
+
+        // start building housingWanted object
+        let housingWantedObject = {
+          AuthorId: account.id,
+          title: request.payload.title,
+          bedrooms: bedrooms,
+          bathrooms: bathrooms,
+          contact: request.payload.contact,
+          body: request.payload.body,
+          status: request.payload.status,
+          details: request.payload.details,
+        };
+
+
+        let housingWantedInstance = await createHousingWanted(housingWantedObject);
+        for (let locationId of request.payload.LocationIds){
+          try {
+            await models.HousingWantedLocation.create({
+              HousingWantedId: housingWantedInstance.id,
+              LocationId: locationId
+            });
+          } catch (error) {
+            console.log(error);
+          }
+        }
```

Comment:
should be wrapped in a conditional checking if LocationIds were set on the payload, since they aren't guaranteed to be.